### ISSUE TITLE
HPA scales up during ScaleDown event

### ISSUE THREAD

**Issue by sarintom** [kubernetes/kubernetes#98174]
**What happened**:

Configured HPA to have `stabilizationWindowSeconds` on `scaleUp` and `scaleDown` events:

```
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: api-server-hpa
spec:
  maxReplicas: 40
  minReplicas: 2
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: api-server-deployment
  metrics:
    - type: Resource
      resource:
        name: cpu
        target:
          type: Utilization
          averageUtilization: 80
  behavior:
    scaleDown:
      stabilizationWindowSeconds: 600
      policies:
        - type: Pods
          value: 2
          periodSeconds: 180
    scaleUp:
      stabilizationWindowSeconds: 300
      policies:
        - type: Pods
          value: 4
          periodSeconds: 30
```

When cpu usage was low a scale down event was triggered but instead it scaled application up since there was recommendations higher than current pod count. And all metrics were below target.

Here are the events:

```
api-server-hpa   Deployment/api-server-deployment   65%/80%    2         40        17         5h20m
api-server-hpa   Deployment/api-server-deployment   65%/80%    2         40        21         5h20m

---

Conditions:
  Type            Status  Reason               Message
  ----            ------  ------               -------
  AbleToScale     True    ScaleDownStabilized  recent recommendations were higher than current one, applying the highest recent recommendation

  Normal  SuccessfulRescale  11m                 horizontal-pod-autoscaler  New size: 17; reason:
  Normal  SuccessfulRescale  10m                 horizontal-pod-autoscaler  New size: 21; reason: All metrics below target

```

**What you expected to happen**:

I expected that scale down wouldn't apply higher recommendations than current pod count.

**How to reproduce it (as minimally and precisely as possible)**:

Add stabilizationWindowSeconds for scaleUp and scaleDown and get recommendation from scaleUp that is higher than current value during that stabilization window. Then trigger scale down.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.19.3 and v1.19.6
- Cloud provider or hardware configuration: AKS

**Comment by arjunrn** [comment]
Fixed with #97348

**Comment by sarintom** [comment]
Ok, thank you for the info. Closing the issue.
